### ISSUE TITLE
Paket prints too many restore information with `--no-restore` in `dotnet build --no-restore`

### ISSUE THREAD

**Issue by nightroman** [fsprojects/Paket#3449]
### Description

Paket prints too many restore information with `--no-restore` in `dotnet build --no-restore`
This is the case with the latest 5.193.0
The older 5.163.2 works as expected/desired.

### Repro steps

Use attached project with latest Paket.5.193.0

Invoke commands:

```cmd
.paket\paket.exe restore
dotnet restore
dotnet build --no-restore
```

### Expected behavior

Output of the last command `dotnet build --no-restore` should be:

```
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Test1 -> C:\Users\rjq\Desktop\_181206_105136_pk\Paket.5.193.0\bin\Debug\netstandard2.0\Test1.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

### Actual behavior

Output of the last command `dotnet build --no-restore` contains too many information about restoring:

```
Microsoft (R) Build Engine version 15.9.20+g88f5fadfbe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Paket version 5.193.0
  The last restore is still up to date. Nothing left to do.
  Performance:
   - Runtime: 128 milliseconds
  Test1 -> C:\Users\rjq\Desktop\_181206_105136_pk\Paket.5.193.0\bin\Debug\netstandard2.0\Test1.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)
```

The picture is even worse if we build a solution with many projects (30+ in our case).
For every project we get 1 useful log line about DLL and 4 noise lines about restoring.

### Known workarounds

I do not know any workarounds. For the moment I prefer to stay with older
Paket.5.163.2 which works as expected. The project is attached, too.

***
- [Paket.5.193.0.zip](https://github.com/fsprojects/Paket/files/2652932/Paket.5.193.0.zip)
- [Paket.5.163.2.zip](https://github.com/fsprojects/Paket/files/2652935/Paket.5.163.2.zip)

**Comment by forki** [comment]
@matthid any ideas about that?

Am Do., 6. Dez. 2018, 12:49 hat Roman Kuzmin <notifications@github.com>
geschrieben:

**Comment by nightroman** [comment]
pinging...

**Comment by forki** [comment]
@matthid I thought it's not calling paket at all if the restore is still up to date?

**Comment by matthid** [comment]
Indeed, if everything is working as it should. Maybe that has been broken again? Even if not there are cases where we cannot "prepare" everything and the paket call on `dotnet restore` still happens. But even in those cases we shouldn't call paket the second time. So something is indeed not working as it should here.

**Comment by rojepp** [comment]
Could this be caused by #3485 setting `PaketRestoreRequired` to true?

**Comment by BlythMeister** [comment]
Yes, it's the same issue.
I have a pr with the fix too.
